feature-libs/cart/base/components/cart-details/cart-details.component.html

```diff
@@ -1,5 +1,6 @@
 <ng-container *ngIf="cart$ | async as cart">
-  <ng-container *ngIf="entries$ | async as entries">
+  <ng-container *ngIf="{entries: entries$ | async, entryGroups: entryGroups$ | async, standaloneEntries: standaloneEntries$ | async, bundle: entries$ | async} as data">
```

Comment:
It seems to me that this is causing your issue with the cart entries still being visible when they are deleted. Having an object as your `ngIf` directive means this is mostly going to resolve as true as long as any of those properties resolve to some non-null value (ie. entries or entryGroups). My first suggestion is to treat these is separate parings so that you don't cross the logic of these objects.

Example:
```
<ng-container *ngIf="entries$ | async as entries">
  // ... all entries code as before stays here
</ng-container>

// could be better to refer to this section as entryGroups than bundles to avoid confusion.
<ng-container *ngFor="let bundle of bundles$ | async">
  // ... all entryGroup code here
</ng-container>
```

To follow up on this suggestion, in future we will want to deprecate `entries` and only use `entryGroups` with `STANDALONE` and `BUNDLE` type groups. We will need to account for supporting both options initially by enabling "feature toggles" to handle any breaking changes. To show you what that means:

```
// The user has enabled feature toggle "isEntryGroupsEnabled" which makes the cart use the new entry groups feature to provide bundle support.
<ng-container [cxFeature]="isEntryGroupsEnabled">
  <ng-container *ngFor="let bundle of bundles$ | async">
    // ... all entryGroup code here with STANDALONE (simple entry) and BUNDLE type items
  </ng-container>
</ng-container>

// The user has NOT enabled feature toggle "isEntryGroupsEnabled" which makes the cart use the OLD entries items. So new features that use entryGroups like bundles will not be supported until the user opts-in.
<ng-container [cxFeature]="!isEntryGroupsEnabled">
  <ng-container *ngIf="entries$ | async as entries">
    // ... all entries code as before stays here
  </ng-container>
</ng-container>
```

With this in mind, try to account for both cases where `entries` logic should stay exactly how it is until we eventually deprecate and remove it. And try to create new logic using the `entryGroups` instead. It is a difficult concept and will be challenging so feel free to ask questions :)

Reply:
![image](https://github.com/SAP/spartacus/assets/153475556/7655a2a7-35a8-4ef0-a2b7-aed44a5658e1)
I tried to change the code and debug it according to the first suggestion, as shown in the picture above, but when the cart has entrygroup and normal products, and delete the normal product, the cart zero issue still exists. You can adapt the code to yours environment and give it a try. BTW, the backend I use is "CX_BASE_URL": "https://40.76.109.9:9002"

https://github.com/SAP/spartacus/assets/153475556/224a4b3e-c7af-4142-b99e-fa36a11731a6

Reply:
Thanks for the supporting material! 

It is clear that there are still `entries` because you have still have added the `entry` from the `entryGroup`. After deleting the last "visible" entry, this entryGroup's entry still exists so nothing will be refreshed. 

That said, I think you need to change the approach as suggested in my previous comment. That is, support `entries` OR `entryGroups` and not both at the same time. Resolving the above issue will not really contribute to the end architecture which is to support one or the other.

You will need to account for STANDALONE products in entryGroups to handle entry-like groups (ie. adding ui for them in the cart). But in this you can avoid using entries all-together just use entryGroups which is what we will prefer to do in the future anyhow.

Reply:
> To follow up on this suggestion, in future we will want to deprecate `entries` and only use `entryGroups` with `STANDALONE` and `BUNDLE` type groups. We will need to account for supporting both options initially by enabling "feature toggles" to handle any breaking changes. To show you what that means:
> 
> ```
> // The user has enabled feature toggle "isEntryGroupsEnabled" which makes the cart use the new entry groups feature to provide bundle support.
> <ng-container [cxFeature]="isEntryGroupsEnabled">
>   <ng-container *ngFor="let bundle of bundles$ | async">
>     // ... all entryGroup code here with STANDALONE (simple entry) and BUNDLE type items
>   </ng-container>
> </ng-container>
> 
> // The user has NOT enabled feature toggle "isEntryGroupsEnabled" which makes the cart use the OLD entries items. So new features that use entryGroups like bundles will not be supported until the user opts-in.
> <ng-container [cxFeature]="!isEntryGroupsEnabled">
>   <ng-container *ngIf="entries$ | async as entries">
>     // ... all entries code as before stays here
>   </ng-container>
> </ng-container>
> ```
Hi Caine，
I'm not sure if I understand you correctly. In the cart-detail component, avoid using **`entries`** OR **`entryGroups`** at the same time?
Based on this principle, I modified the original code and did some debugging locally
Previously (left), the **getBundleEntryGroups** method obtained an Observable<OrderEntryGroup[]> containing only `CONFIGURABLEBUNDLE` type;
Now (right), it has become an Observable<OrderEntryGroup[]> containing `STANDALONE` type(normal product).
When entrygroup type is `STANDALONE`, give it's` entryGroup.entries` to a `Observable<OrderEntry[]> `as before, use to render normal cart list as snapshot2.
Because the code for getting `entries` is commented out，in this way, all `entries`' comes from `entrygroups` (including normal product).

But the cart zero issue is still existing. 




![image](https://github.com/SAP/spartacus/assets/153475556/c13dc0b1-f798-4eff-8b07-b456df269152)
![image](https://github.com/SAP/spartacus/assets/153475556/b193cf4c-1af1-4c18-a7c9-aa0a0e87dba0)

Reply:
Hi Caine,
Here are 2 updates:
1. I followed your previous comments and all entries come from entrygroups. 
**The cart zero issue has been fixed.**
2. **The hierarchy-related components have been put into shared/component**, and template outlets were used to solve the code dependency problem.

Please review the code when you're available. Looking forward to your comments.

Reply:
Thanks for the updates! Its looking much cleaner now and you seem to have understood the concepts well :) I won't go into a great deal of comments now since this PR is still in progress and I have no major concerns with the course it is taking now. 

Just one minor remark that you should try to move the styling inputs into scss and use a selector like `nth` to nest paddings. We should try avoid styling logic in our class files unless we really need to to keep it clean.

Reply:
Thanks for your updates.
Of course, I will try to move the styling into scss in the next update. 
As for using a selector like nth to nest paddings, because the tree does not know how many layers it has before rendering, it does not look very simple. Anyway, when the other pages are almost developed, I will take the time to check it.
Thanks